### PR TITLE
chore: fix graphql user for boarding

### PR DESCRIPTION
**Pull request by 120EE0692**
### Solved or Linked Issues

Reference any and all issues solved and related to this pull request here. Mention if there are none.

### Description and Changes

Please describe your pull request with all changes made.

### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../README.md) to this repository.

- [ ] Make sure you are requesting to **pull a work/topic/feature/bugfix branch** (right side). Don't request your default branch!
- [ ] Make sure you are making a pull request against the **development** (left side). Also you should start _your branch_ off _default branch_ and follow the branch naming guidelines.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation and mentioned the related issues (if appropriate)
- [ ] I have added the appropriate labels, requested a review from a repository manager and assigned myself to this pull request.

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../blob/development/CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: server/schema/user/user.datasources.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ const create = async (uid, fullName, email, interestedTopics, session, authToken
     );
 
     await admin.auth().setCustomUserClaims(uid, {
-      mid: _user.id,
+      mid: _user[0].id,
```
Comment (rutajdash) [inline server/schema/user/user.datasources.js:137]: Does this fix the missing mid issue?

Comment (rutajdash) [inline server/schema/user/user.datasources.js:137]: Does this fix the missing mid issue?

File: server/schema/user/user.datasources.js [hunk 2]
```diff
@@ -152,6 +152,9 @@ const create = async (uid, fullName, email, interestedTopics, session, authToken
 };
 
 // TODO: Update all redundancies
+
+const addStorePath = (uid, storePath) => admin.auth().updateUser(uid, { picture: { photoURL: storePath } });
```
Comment (rutajdash) [inline server/schema/user/user.datasources.js:156]: Rename this to `updateProfilePicture` and modify the function to support changing the full picture object.

File: server/schema/user/user.model.js [hunk 3]
```diff
@@ -79,9 +79,8 @@ const UserSchema = new Schema(
     },
     interestedTopics: [
       {
-        type: Number,
+        type: String,
```
Comment (rutajdash) [inline server/schema/user/user.model.js:82]: The topics we add are all categories and subcategories, right? Why change from number to string then?

File: server/schema/user/user.resolver.js [hunk 4]
```diff
@@ -43,9 +43,7 @@ const canUpdateUser = (id, mid, session, authToken, decodedToken, fieldNodes, ne
   if (needsAdmin && !UserPermission.exists(session, authToken, decodedToken, 'user.write.all')) {
     throw APIError('FORBIDDEN', null, { reason: 'The user does not have the required administrative priveledges.' });
   } else if (
-    (mid === id &&
-      !UserPermission.exists(session, authToken, decodedToken, 'user.write.self') &&
-      !UserPermission.exists(session, authToken, decodedToken, 'user.write.all')) ||
+    (mid === id && !UserPermission.exists(session, authToken, decodedToken, 'user.write.self')) ||
```
Comment (rutajdash) [inline server/schema/user/user.resolver.js:46]: Why this change?

File: server/schema/user/user.resolver.js [hunk 5]
```diff
@@ -249,7 +247,14 @@ module.exports = {
...
   },
-
+  addUserPicStorePath: async (_parent, { id, storePath }, { mid, session, authToken, decodedToken, API: { User } }) => {
+    try {
+      canUpdateUser(id, mid, session, authToken, decodedToken);
+      return addStorePath(id, storePath, mid, session, authToken, decodedToken, User);
+    } catch (error) {
+      throw APIError(null, error);
+    }
+  },
```
Comment (rutajdash) [inline server/schema/user/user.resolver.js:257]: Refactor to `updateUserProfilePicture`

File: server/schema/user/user.resolver.js [hunk 6]
```diff
@@ -343,7 +348,7 @@ module.exports = {
       const _user = await User.updateDetails(id, { interestedTopics }, session, authToken, mid);
       return _user;
     } catch (error) {
-      throw APIError(null, error);
+      throw APIError(null, error, 'failed to update user topics');
```
Comment (rutajdash) [inline server/schema/user/user.resolver.js:351]: Why was this change required?

**Review by rutajdash — CHANGES_REQUESTED** [review]

File: server/schema/user/user.datasources.js (hunk 1 above)
Comment (120EE0692) [reply server/schema/user/user.datasources.js:137]: Yes, this fix that issue.

File: server/schema/user/user.resolver.js (hunk 4 above)
Comment (120EE0692) [reply server/schema/user/user.resolver.js:46]: This basically checking the user can updates some thing from own account or not. user.write.self is sufficient  to do that. And super admin always have that role.
